src/lib.rs

```diff
@@ -1,62 +1,143 @@
             }
-        },
-        _ => {
-            return Err(Error::from_raw(err));
+            Ok(result as *const T)
         }
-    };
+
+        Ok(Context {
+            system: load(sys::IVRSystem_Version)?,
+            compositor: load(sys::IVRCompositor_Version)?,
+        })
+    }
+
+    pub fn system(&self) -> System { unsafe { System(&*self.system) } }
+    pub fn compositor(&self) -> Compositor { unsafe { Compositor(&*self.compositor) } }
 }
 
-/// Shutdowns all openvr related systems
-pub fn shutdown() {
-    unsafe {
-        openvr_sys::VR_ShutdownInternal();
+impl Drop for Context {
+    fn drop(&mut self) {
+        unsafe { sys::VR_ShutdownInternal() }
+        INITIALIZED.store(false, Ordering::AcqRel);
     }
 }
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum ApplicationType {
+    /// Some other kind of application that isn't covered by the other entries 
+    Other = sys::EVRApplicationType_EVRApplicationType_VRApplication_Other as isize,
+    /// Application will submit 3D frames
+    Scene = sys::EVRApplicationType_EVRApplicationType_VRApplication_Scene as isize,
+    /// Application only interacts with overlays
+    Overlay = sys::EVRApplicationType_EVRApplicationType_VRApplication_Overlay as isize,
+    /// Application should not start SteamVR if it's not already running, and should not keep it running if everything
+    /// else quits.
+    Background = sys::EVRApplicationType_EVRApplicationType_VRApplication_Background as isize,
+    /// Init should not try to load any drivers. The application needs access to utility interfaces (like IVRSettings
+    /// and IVRApplications) but not hardware.
+    Utility = sys::EVRApplicationType_EVRApplicationType_VRApplication_Utility as isize,
+    /// Reserved for vrmonitor
+    VRMonitor = sys::EVRApplicationType_EVRApplicationType_VRApplication_VRMonitor as isize,
+    /// Reserved for Steam
+    SteamWatchdog = sys::EVRApplicationType_EVRApplicationType_VRApplication_SteamWatchdog as isize,
+    /// Start up SteamVR
+    Bootstrapper = sys::EVRApplicationType_EVRApplicationType_VRApplication_Bootstrapper as isize,
+}
+
+pub struct InitError(sys::EVRInitError);
+
+impl fmt::Debug for InitError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let msg = unsafe {
+            CStr::from_ptr(sys::VR_GetVRInitErrorAsSymbol(self.0))
+        };
+        f.pad(msg.to_str().expect("OpenVR init error symbol was not valid UTF-8"))
+    }
+}
+
+impl error::Error for InitError {
+    fn description(&self) -> &str {
+        let msg = unsafe {
+            CStr::from_ptr(sys::VR_GetVRInitErrorAsEnglishDescription(self.0))
+        };
+        msg.to_str().expect("OpenVR init error description was not valid UTF-8")
+    }
+}
+
+impl fmt::Display for InitError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.pad(error::Error::description(self))
+    }
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum Eye {
```

Comment:
I actually like that they were in the common module since they are also used in some other interfaces.